lib/ansible/cli/galaxy.py

```diff
@@ -1451,13 +1458,18 @@ def execute_list_collection(self, artifacts_manager=None):
                     display.vvv("No collections found at {0}".format(collection_path))
                     continue
 
-                # Display header
-                fqcn_width, version_width = _get_collection_widths(collections)
-                _display_header(collection_path, 'Collection', 'Version', fqcn_width, version_width)
+                if output_format in ('yaml', 'json'):
+                    collections_in_paths[collection_path] = [
```

Comment:
Why add another level of nesting with a list? Just have a dict comprehension.

Reply:
In case we ever add more field to this like author name or something of the sort. Keeping it as a dict entry in a list gives us more flexibility in the future.

Reply:
After further consideration, I actually agree with this point. It makes looping over this data easier too, as seen in the tests. If we wanted to add more information to the sub-dictionary in the future, that's possible too. Thanks for this feedback.

Reply:
Well, if we want flexibility, we could have a dict value but I still expect it to we inside a dict, not inside a list.
It's `{'ns.coll': {'fqcn': 'ns.coll', 'ver': '1.2.0'}}` vs `[{'fqcn': 'ns.coll', 'ver': '1.2.0'}]`: with a list, it's harder to access a specific collection as you'd have to do a full list scan to find in which element your target is while with a dict you can access it by key instantly.

cc @shanemcd

Reply:
Oh, I see it's addressed now.